src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -40,9 +40,7 @@
 class PylintFieldSet(LinterFieldSet):
     required_fields = (PythonSources,)
 
-    sources: PythonSources
     dependencies: Dependencies
-    compatibility: PythonInterpreterCompatibility
```

Comment:
Both the removed fields aren't directly used in the `FieldSets` parameter because we end up re-resolving the original `Target`s. We could keep them here as a signal, but that seemed noisy. 

The class definition now expresses the semantics:
* You must have `PythonSources` to be lintable
* We will directly access the `Dependencies` field
* Other fields might be used in the rule body by us re-resolving the `Target`.